google/cloud/ndb/key.py

```diff
@@ -381,21 +383,25 @@ def __eq__(self, other):
     def __lt__(self, other):
         """Less than ordering."""
         if not isinstance(other, Key):
-            return NotImplemented
+            raise TypeError
```

Comment:
I thought NotImplemented was also the protocol for Python 2 comparison functions. Am I mistaken?

Reply:
Not sure about the protocol, but it just didn't work. The NotImplemented was being evaluated as an expression and returned a True/False value instead of raising.